### PR TITLE
Adds an account settings management helper object

### PR DESCRIPTION
**Pull request by kernkw**
This PR should simplify the ability to retrieve or update account settings with just the name of the setting, instead of needing to know if a setting is a tracking, mail, user, or partner. This PR also pulls in rspec and faker gem to increase test thoroughness. 

See example or readme for example.

Let me know your thoughts on this, I also have plans to add wrappers around other account management endpoints, such as subuser creation and management, stats, etc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,27 @@ puts response.status_code
 puts response.body
 puts response.headers
 ```
+## Manage Account Settings
```
Comment (thinkingserious) [inline README.md]: This should go in the README in the helper file, see the mail helper for an example: https://github.com/sendgrid/sendgrid-ruby/tree/master/lib/sendgrid/helpers/mail. Then you can link to it from here.

File: examples/settings/settings.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require 'sendgrid-ruby'
```
Comment (thinkingserious) [inline examples/settings/settings.rb]: Please move this to examples/helpers/settings

File: lib/sendgrid/settings/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+**This module allows you to quickly and easily build a Settings object for retreiving or updating you SendGrid Settings.**
```
Comment (thinkingserious) [inline lib/sendgrid/settings/README.md]: Please move to the helpers directory

File: lib/sendgrid/settings/mail_settings.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module SendGrid
```
Comment (thinkingserious) [inline lib/sendgrid/settings/mail_settings.rb]: Please move to the helpers directory.

File: lib/sendgrid/settings/partner_settings.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+module SendGrid
```
Comment (thinkingserious) [inline lib/sendgrid/settings/partner_settings.rb]: Please move to the helpers directory

File: lib/sendgrid/settings/settings.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
+require_relative 'mail_settings'
```
Comment (thinkingserious) [inline lib/sendgrid/settings/settings.rb]: Please move to the helpers directory

File: lib/sendgrid/settings/user_settings.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+module SendGrid
```
Comment (thinkingserious) [inline lib/sendgrid/settings/user_settings.rb]: Please move to the helpers directory

File: spec/sendgrid/settings/mail_settings_spec.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
+require 'spec_helper'
```
Comment (thinkingserious) [inline spec/sendgrid/settings/mail_settings_spec.rb]: Please move to the helpers directory

File: spec/sendgrid/settings/partner_settings_spec.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
+require 'spec_helper'
```
Comment (thinkingserious) [inline spec/sendgrid/settings/partner_settings_spec.rb]: Please move to the helpers directory

File: spec/sendgrid/settings/settings_spec.rb [hunk 10]
```diff
@@ -0,0 +1,25 @@
+require 'spec_helper'
```
Comment (thinkingserious) [inline spec/sendgrid/settings/settings_spec.rb]: Please move to the helpers directory

File: spec/sendgrid/settings/tracking_settings_spec.rb [hunk 11]
```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
```
Comment (thinkingserious) [inline spec/sendgrid/settings/tracking_settings_spec.rb]: Please move to the helpers directory

File: spec/sendgrid/settings/user_settings_spec.rb [hunk 12]
```diff
@@ -0,0 +1,24 @@
+require 'spec_helper'
```
Comment (thinkingserious) [inline spec/sendgrid/settings/user_settings_spec.rb]: Please move to the helpers directory

**Comment by thinkingserious** [comment]
Very nice @kernkw!

We just need some things to be moved around, please see the comments. Generally, please follow the conventions that the mail helper uses. Thanks!

File: README.md (hunk 1 above)
Comment (kernkw) [inline README.md]: Cool makes sense, any preference on how I link to it from the main README? I didnt see a link to the mail helper there currently

Comment (thinkingserious) [inline README.md]: The current links are in-line like so: 
- https://github.com/sendgrid/sendgrid-ruby/blob/master/examples/mail/mail.rb#L29
- https://github.com/sendgrid/sendgrid-ruby/blob/master/USAGE.md#v3-mail-send

But perhaps we can add a helpers section in the README too.

**Comment by kernkw** [comment]
@thinkingserious it looks like this build is failing because its not exporting the env variables from the repo settings, my guess is that this because my pr is coming from a fork so when it builds the pr its not picking up those variables, i could be wrong but thats a theory

This line is missing from my PR build and thus all of the tests are failing with 403 status code
```
Setting environment variables from repository settings
$ export MOCK_HOST=[secure]
```

**Comment by thinkingserious** [comment]
@kernkw Correct, that is a known issue that will be fixed soon; meanwhile, you can test locally like this: https://github.com/sendgrid/sendgrid-ruby/pull/72#issuecomment-231172405 Thanks!

**Comment by kernkw** [comment]
@thinkingserious i am getting some test errors running the prism server, followed instructions and looked at the prism json i am not sure why the error is occurring, i pryed into the code to see what api call was failing.

Basically i looks like prism is not reading integer params as integers but is receiving them as strings.
I tested in sendgrid prod account and this call passes so i assume its something due to prism endpoints.

Not sure what i am doing differently from you, i downloaded latest prism and pulled a fresh master branch from sendgrid-ruby

```
prism run --mock --list --spec https://api.stoplight.io/v1/versions/563a5309daad691100fb05af/export/stoplight.json

curl http://localhost:4010/v3/geo/stats\?end_date\=2016-04-01\&country\=US\&aggregated_by\=day\&start_date\=2016-01-01\&limit\=1\&offset\=1
{
    "errors": [
        {
            "context": "request.url.queryString",
            "description": "Invalid type. Expected: [array,integer], given: string",
            "field": "limit",
            "location": "limit",
            "type": "invalid_type"
        },
        {
            "context": "request.url.queryString",
            "description": "Invalid type. Expected: [array,integer], given: string",
            "field": "offset",
            "location": "offset",
            "type": "invalid_type"
        }
    ]
}
```

but when i remove the limit and offset param it works

```
curl http://localhost:4010/v3/geo/stats\?end_date\=2016-04-01\&country\=US\&aggregated_by\=day\&start_date\=2016-01-01
[
  {
    "date": "2015-10-11",
    "stats": [
      {
        "metrics": {
          "clicks": 0,
          "opens": 0,
          "unique_clicks": 0,
          "unique_opens": 0
        },
        "name": "TX",
        "type": "province"
      }
    ]
  },
```

**Comment by thinkingserious** [comment]
@kernkw,

I just tested your branch on my machine and that test passed.

Can you please try these steps?

https://github.com/sendgrid/sendgrid-ruby/pull/74#issuecomment-231548757

Thanks!

**Comment by kernkw** [comment]
@thinkingserious ok cool it works thanks, i figured their website would download the latest version of prism but they have you download v0.1.2 so must have fixed the issue with their latest.

I am gonna add rspec tests to run in the rake task, so travis CI can pick those up, sound good?

**Comment by thinkingserious** [comment]
Sounds good, thanks!

**Comment by thinkingserious** [comment]
Hi @kernkw,

This one finally rose up to the top of the queue. Could you please fix the conflicts so that I may test and merge? Thanks!

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=70) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=70) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-ruby?pullRequest=70) it.</sub>

**Pull request merged**